src/helpers/getByAPI.js

```diff
@@ -56,6 +56,17 @@ const getNodeByText = (node, text) => {
           : text.test(textToTest);
       }
     }
+
+    const isParentTextComponent = filterNodeByType(node.parent?.parent, Text);
```

Comment:
checking parents non-recursively will result in a failure in e.g. this scenario:
```js
const Wrap = ({ children }) => <>{children}</>;

  const CustomText = () => {
    return <React.Fragment>Hello World</React.Fragment>;
  };

  expect(
    render(
      <Text>
        <Wrap>
          <CustomText />
        </Wrap>
      </Text>
    ).queryByText('Hello World')
  ).toBeTruthy();
```

Reply:
Please check this PR as well: https://github.com/callstack/react-native-testing-library/pull/554 as it's changing the way we're interfacing text to be similar to React Testing Library. Grabbing by text with non-direct string children is tricky and we'll likely need to trade-off some functionality, or come up with a more complex solution.

We want to solve this problem once for good, so if you're willing to help improve the library, please chime in there and see how we can support this scenario with string precision API

Reply:
FYI, #554 is now merged. Would you be still interested in pursuing this improvement? 😊